### PR TITLE
[2607-DEV-472] Add test infrastructure: Vitest + CI test job + first unit tests

### PR DESCRIPTION
**Pull request by teamenjoyvd**
Closes #472

## Summary
- Adds `vitest` + `vite-tsconfig-paths` devDependencies and a `test` script (`vitest run`); `test:mock` untouched.
- `vitest.config.ts`: `environment: 'node'`, tsconfig-paths plugin for the `@/` alias, `include: ['lib/**/*.test.ts']`.
- `lib/public-routes.test.ts`: asserts the 16-entry `PUBLIC_ROUTE_PATTERNS` guest surface exactly, and `isPublicRoute` (via `NextRequest` fixtures) against every public path plus `/profile`, `/api/profile`, `/api/admin/members`.
- `lib/format.test.ts`: fixed-instant cases for `formatDate`/`formatTime`/`formatDateTime`/`formatCurrency`; `toSofiaLocalInput` ↔ `fromSofiaLocalInput` round-trips around the 2026-03-29 and 2026-10-25 EU DST boundaries; `timeAgoMs` boundaries (59s/60s, 59m/60m, 23h/24h, negative diff); `excerptFromJSONContent` (null input, mark-split sibling text nodes, block-spacing, 160-char truncation).
- `.github/workflows/ci.yml`: new `test` job (same node-22/npm-install pattern as the other jobs).

## Notes for reviewers
- `formatDate`/`formatDateTime` expectations include the bg-BG " г." year suffix and `formatCurrency` expects a non-breaking space (U+00A0) before "€" — both are the actual `Intl` output on this locale/ICU, not typos.
- DST round-trip fixtures sit a few hours clear of the exact transition instant rather than adjacent to it: `fromSofiaLocalInput`'s offset guess-and-correct can misfire by exactly one hour when the naive-as-UTC reinterpretation of the local string lands on the other side of the transition from the original instant. NOTED (not done, out of DoD scope): this is a pre-existing edge-case in `lib/format.ts:131` `fromSofiaLocalInput` for input timestamps within roughly ~2h of a DST transition instant — not fixed here since the ticket scope is test infrastructure only.
- All tests are mock-free pure-function tests — no jsdom, no Supabase/Clerk mocking. No UI touched, so the 390px constraint doesn't apply.

## Test plan
- [x] `npx vitest run` — 45/45 passing
- [x] `npm run check-types` — clean
- [x] `npx eslint` on new files — clean
- [x] CI green on this PR (Type Check, Lint, Test, Build, Security Audit all passed)
- [x] Vercel preview READY ("Deployment has completed")

## Session State
**Status:** DONE
**Agent Type:** Claude
**Completed:**
- [x] vitest + vite-tsconfig-paths installed, `test` script added
- [x] `vitest.config.ts` created
- [x] `lib/public-routes.test.ts` written and passing
- [x] `lib/format.test.ts` written and passing
- [x] CI `test` job added
- [x] Local verification: vitest (45/45), tsc --noEmit, eslint all clean
- [x] CI green on PR (Type Check, Lint, Test, Build, Security Audit)
- [x] Vercel preview deployment completed
**Next:** None — ready for human review and merge.

## Summary by CodeRabbit

* **Bug Fixes**
  * Expanded automated test coverage for date/time formatting, currency display, timezone handling, relative time, content excerpts, and public route behavior.
  * Added CI test execution so code changes are verified automatically on every run, helping catch regressions earlier.

### CODE REVIEW

File: lib/format.test.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+describe('formatDateTime', () => {
+  it('formats as DD.MM.YYYY, HH:mm in Europe/Sofia', () => {
+    expect(formatDateTime(FIXED_ISO)).toBe('18.03.2026 г., 14:00')
+  })
+})
+
+describe('formatCurrency', () => {
+  // de-DE inserts a non-breaking space (U+00A0) before the currency symbol.
+  it('formats EUR with de-DE grouping/decimal separators', () => {
+    expect(formatCurrency(1234)).toBe('1.234,00 €')
```
[the remark this answers is not in the document]
Comment (teamenjoyvd) [reply lib/format.test.ts]: Applied — switched to the explicit `\u00A0` escape.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lib/format.test.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  })
+
+  it('formats zero correctly', () => {
+    expect(formatCurrency(0)).toBe('0,00 €')
```
[the remark this answers is not in the document]
Comment (teamenjoyvd) [reply lib/format.test.ts]: Applied — switched to the explicit `\u00A0` escape.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: lib/public-routes.test.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      '/api/webhooks/(.*)',
+      '/api/calendar',
+      '/api/calendar/(.*)',
+      '/api/events/:id',
+      '/api/socials',
+      '/api/socials/(.*)',
+    ])
+    expect(PUBLIC_ROUTE_PATTERNS).toHaveLength(16)
+  })
+})
```
[the remark this answers is not in the document]
Comment (teamenjoyvd) [reply lib/public-routes.test.ts:27]: Skipping — this assertion is a direct DoD item from issue #472 ("asserts PUBLIC_ROUTE_PATTERNS equals the documented 16-entry guest surface exactly"), and the ticket frames it as the security control itself: the ticket's own stated leverage is that a bad edit to the guest-route list is either a lockout or an unintentionally public route, and only an equality assertion (not `isPublicRoute` behavioral tests) catches an *added* pattern that nobody intended. Keeping it.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
